lesson_6/Ioan_Juglea/Homework.py

```diff
@@ -0,0 +1,96 @@
+    ('IE', ['78 ', '81 ', '82 ', '82 ', '85 ', '87 ', '88 ', '89 ', '91 ']),
+    ('IS', ['93 ', '95 ', '96 ', '96 ', ': ', ': ', '98 ', '99 ', '98 ']),
+    ('IT', ['62 ', '63 ', '69 ', '73 ', '75 ', '79 ', '81 ', '84 ', '85 ']),
+    ('LT', ['60 ', '60 ', '65 ', '66 ', '68 ', '72 ', '75 ', '78 ', '82 ']),
+    ('LU', ['91 ', '93 ', '94 ', '96 ', '97 ', '97 ', '97 ', '93 b', '95 ']),
+    ('LV', ['64 ', '69 ', '72 ', '73 ', '76 ', '77 b', '79 ', '82 ', '85 ']),
+    ('ME', [': ', '55 ', ': ', ': ', ': ', ': ', '71 ', '72 ', '74 ']),
+    ('MK', [': ', '58 ', '65 ', '68 ', '69 ', '75 ', '74 ', '79 ', '82 ']),
+    ('MT', ['75 ', '77 ', '78 ', '80 ', '81 ', '81 ', '85 ', '84 ', '86 ']),
+    ('NL', ['94 ', '94 ', '95 ', '96 ', '96 ', '97 ', '98 ', '98 ', '98 ']),
+    ('NO', ['92 ', '93 ', '94 ', '93 ', '97 ', '97 ', '97 ', '96 ', '98 ']),
+    ('PL', ['67 ', '70 ', '72 ', '75 ', '76 ', '80 ', '82 ', '84 ', '87 ']),
+    ('PT', ['58 ', '61 ', '62 ', '65 ', '70 ', '74 ', '77 ', '79 ', '81 ']),
+    ('RO', ['47 ', '54 ', '58 ', '61 b', '68 ', '72 ', '76 ', '81 ', '84 ']),
+    ('RS', [': ', ': ', ': ', ': ', '64 ', ': ', '68 ', '73 ', '80 ']),
+    ('SE', ['91 ', '92 ', '93 ', '90 ', '91 ', '94 b', '95 ', '93 ', '96 ']),
+    ('SI', ['73 ', '74 ', '76 ', '77 ', '78 ', '78 ', '82 ', '87 ', '89 ']),
+    ('SK', ['71 ', '75 ', '78 ', '78 ', '79 ', '81 ', '81 ', '81 ', '82 ']),
+    ('TR', [': ', '47 ', '49 ', '60 ', '70 ', '76 ', '81 ', '84 ', '88 ']),
+    ('UK', ['83 ', '87 ', '88 ', '90 ', '91 ', '93 ', '94 ', '95 ', '96 ']),
+    ('XK', [': ', ': ', ': ', ': ', ': ', ': ', '89 ', '93 ', '93 ']),
+]
+
+def prepare_Data(dataset):
+    refined_Data = dict()
+    for country, coveragelist in dataset:
+        coverage_per_country = []
+        for index, coverage in reversed(list(enumerate(coveragelist))):
+            if coverage != ': ':
+                year = 2011 + index
+                coverage_per_country.append(
+                    {
+                        'year': str(year),
+                        'coverage': int(coverage[:-1])
+                    }
+                )
+        refined_Data.update({country: coverage_per_country})
+    return refined_Data
+
+def retrieve_data_for_year(dataset, year):
+    refined_Data = prepare_Data(dataset)
+    new_Data = []
+    for country in refined_Data:
+        for dictionary in refined_Data[country]:
+            if dictionary['year'] == year:
+                new_Data.append({country: dictionary['coverage']})
+    final_Data = {int(year): new_Data}
+    return final_Data
+
+def retrieve_data_for_country(dataset, chosen_country):
+    refined_Data = prepare_Data(dataset)
+    for country in refined_Data:
+        if country == chosen_country:
+            new_Data = refined_Data[country]
+    final_Data = {chosen_country: new_Data}
+    return final_Data
+
+def perform_average(dataset):
+    suma = 0
+    count = 0
+    for key in dataset:
```

Comment:
ii fain ca functia de average suporta mai multe tipuri de imput. Un sfat aici ar fi sa pregatesti listele pentru suma in functie de ce primesti si dupaia sa folosesti `sum()` https://docs.python.org/3/library/functions.html#sum